.github/workflows/kubeflow-release.yaml

```diff
@@ -134,10 +134,10 @@ jobs:
           echo "pr_merged=false" >> $GITHUB_OUTPUT
           exit 1
 
-  # 5. Create a release (Mock-Up workflow it will be fullfill by RHOAIENG-15391)
+  # 5. Create a release 
   create-release:
     needs: [update-release-images, check-pr-merged]
     if: ${{ needs.check-pr-merged.outputs.pr_merged == 'true' && inputs.create-new-release == 'true' }}
     uses: opendatahub-io/kubeflow/.github/workflows/create-release.yaml@main
```

Comment:
will (unfortunately) need to defer to @atheo89 here to see if she had any strong convictions for setting it up this way.

Reply:
We need to specify the branch from where the GitHub looks for the `create-release.yaml` file. If you change @main to another branch or tag, GitHub will attempt to find the file in the specified branch or tag. Moreover, If the file exists in another branch but has differences (e.g., parameters, syntax, or behavior), the workflow may fail due to incompatibilities.

Reply:
Exactly, incompatibility is my concern. Can we be sure that the version of the workflow that is sitting on `main` will always be compatible with all the code along the commit tree? If we ever need to do a release from an old commit, there could be some incompatibility because the workflow on `main` might have evolved differently than what was expected at the time of the commit.

If we don't freeze the branch with `@main`, the workflow will be taken from the checked-out code, which, I assume, will always be compatible.

Reply:
Yes, In general, we maintain consistency in our GitHub Actions workflow across branches. Typically, the workflows present on the main branch are propagated to all other branches as well.
However, removing @main from the reference will cause issues because GitHub would no longer know which branch to pull the callable action from.

Reply:
That's right. We need to identify the reference since the call is a step, not a job.
And since everything is synced across branches, we shouldn't face incompatibility issues.
Thanks for the clarification!